matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/permalinks/PermalinkParser.kt

```diff
@@ -21,20 +21,22 @@ import android.net.UrlQuerySanitizer
 import org.matrix.android.sdk.api.MatrixPatterns
 
 /**
- * This class turns an uri to a [PermalinkData]
+ * This class turns a uri to a [PermalinkData]
+ * element-based domains (e.g. https://app.element.io/#/user/@chagai95:matrix.org) permalinks or matrix.to permalinks (e.g. https://matrix.to/#/@chagai95:matrix.org)
+ * 
  */
 object PermalinkParser {
 
     /**
-     * Turns an uri string to a [PermalinkData]
+     * Turns a uri string to a [PermalinkData]
      */
     fun parse(uriString: String): PermalinkData {
         val uri = Uri.parse(uriString)
         return parse(uri)
     }
 
     /**
-     * Turns an uri to a [PermalinkData]
+     * Turns a uri to a [PermalinkData]
      */
     fun parse(uri: Uri): PermalinkData {
         if (!uri.toString().startsWith(PermalinkService.MATRIX_TO_URL_BASE)) {
```

Comment:
You won't get to you change, uri has to start by `https://matrix.to/#/`

Reply:
I changed the MATRIX_TO_URL_BASE variable on my fork to be the element-based domain permalink, that's why for me it worked, I think this is easier to find this issue than to find a hard coded array, I could add this to a comment but it will of course be better to make a variable out of this, I don't think I can get to doing that easily but if anyone else wants to pick this PR and do it before I do, please do... I understand if you don't want to merge like this, I just wanted others to not spend a day looking for this issue 🙈 on our fork it's working now...

Reply:
Ah, [we build matrix.to links](https://github.com/vector-im/element-android/blob/main/vector/src/main/java/im/vector/app/features/link/LinkHandlerActivity.kt#L105), so it's OK I think...

Reply:
For you it's fine but other forks use element-based domain permalinks 🙈

Reply:
Ok, thanks for the explaination. We definitely need to improve this code, and add unit test.

Also I guess you will have the same problem for room URL, for instance:
> https://develop.element.io/#/room/#element-android:matrix.org

Reply:
yep thanks, we don't have room links but thx for the heads up 😉